src/network/os1/client/prot/game/GameClientRepository.ts

```diff
@@ -30,5 +63,31 @@ export default class GameClientRepository extends ClientRepository {
         this.bind(new ClientCheatDecoder(), new ClientCheatHandler())
         this.bind(new IfButtonDecoder(), new IfButtonHandler())
         this.bind(new ClanDecoder(), new ClanHandler())
+        this.bind(new OpPlayer1Decoder(), new OpPlayerHandler())
```

Comment:
Much better. You could do the same for the decoders to reduce some boilerplate too! Pass op and opcode into "OpPlayerDecoder":

```ts
this.bind(new OpPlayerDecoder(1, 245), new OpPlayerHandler());
this.bind(new OpPlayerDecoder(2, 146), new OpPlayerHandler());
this.bind(new OpPlayerDecoder(3, 102), new OpPlayerHandler());
this.bind(new OpPlayerDecoder(4, 78), new OpPlayerHandler());
```

```ts
export default class OpPlayerDecoder extends MessageDecoder {
    size = 2;
    limit = GameClientLimit.CLIENT;

    constructor(op: number, opcode: number) {
        this.op = op;
        this.opcode = opcode;
    }

    read(buf: Packet): ClientMessage {
        let playerId = -1;
        if (this.op === 1) {
            playerId = buf.g2();
        } else if (this.op === 2) {
            playerId = buf.g2();
        } else if (this.op === 3) {
            playerId = buf.g2_alt1();
        }

        return new OpPlayer(this.op, playerId);
    }
}
```